bin/bexley/make-bexley-ww-postcode-db

```diff
@@ -214,22 +204,24 @@ while( my $row = $csv->getline($fh) ) {
     } elsif ( $record_id == $id_blpu ) {
         @row_h{@columns_blpu} = @$row;
 
-        next if ( $row_h{LOGICAL_STATUS} || 0 ) != 1;
-        next unless $row_h{PARENT_UPRN};
-
-        $query_cu->execute(
-            $row_h{UPRN},
-            $row_h{PARENT_UPRN},
-        );
+        $uprn_extra_data{ $row_h{UPRN} }{has_parent} = !!$row_h{PARENT_UPRN};
+        $uprn_extra_data{ $row_h{UPRN} }{blpu_class} = $row_h{BLPU_CLASS};
     } elsif ( $record_id == $id_lpi ) {
         @row_h{@columns_lpi} = @$row;
 
         next if ( $row_h{LOGICAL_STATUS} || 0 ) != 1;
         next unless $row_h{POSTCODE};
 
+        # 'P' or 'PP' mean the property is a 'parent shell' and so should not be
+        # offered as a selectable address.
+        my $blpu_class = $uprn_extra_data{ $row_h{UPRN} }{blpu_class} // '';
+        next if $blpu_class && ( $blpu_class eq 'P' || $blpu_class eq 'PP' );
```

Comment:
Doesn't need the first if I think, given you've made sure it's not undefined in the line before.